md/enhanced.py

```diff
@@ -351,3 +351,98 @@ def sample_from_log_weights(weighted_samples, log_weights, size):
     assert np.abs(np.sum(weights) - 1) < 1e-5
     idxs = np.random.choice(np.arange(len(weights)), size=size, p=weights)
     return weighted_samples[idxs]
+
+
+from fe import free_energy
```

Comment:
Do we care about these being at the top of the file?

Reply:
yes, will move soon!